extensions/analyticsdx-vscode-templates/test/vscode-integration/templateEditing/layout.test.ts

```diff
@@ -268,36 +269,56 @@ describe('TemplateEditorManager configures layoutDefinition', () => {
-    expect(node, 'pages[0].layout.center.items[3].name propNode').to.be.not.undefined;
+  async function testHover(
+    doc: vscode.TextDocument,
+    uri: vscode.Uri,
+    tree: JsonNode,
+    jsonpath: JSONPath,
+    expectedHoverText: string
+  ) {
+    const node = tree && findNodeAtLocation(tree, jsonpath)?.parent;
+    const jsonpathStr = jsonPathToString(jsonpath);
+    expect(node, `${jsonpathStr} propNode`).to.be.not.undefined;
     const nameNode = node!.children?.[0];
-    expect(nameNode, 'nameNode').to.not.be.undefined;
+    expect(nameNode, `${jsonpathStr} nameNode`).to.not.be.undefined;
     let hovers = await getHovers(uri, doc.positionAt(nameNode!.offset));
-    expect(hovers, 'nameNode hovers').to.not.be.undefined;
+    expect(hovers, `${jsonpathStr} nameNode hovers`).to.not.be.undefined;
     // on the name field, it should just return the hover from the schema
-    expect(hovers.length, 'nameNode hovers.length').to.equal(1);
+    expect(hovers.length, `${jsonpathStr} nameNode hovers.length`).to.equal(1);
 
     const valueNode = node!.children?.[1];
-    expect(valueNode, 'valueNode').to.not.be.undefined;
+    expect(valueNode, `${jsonpathStr} valueNode`).to.not.be.undefined;
     hovers = await waitFor(
       () => getHovers(uri, doc.positionAt(valueNode!.offset)),
       hovers => hovers.length >= 2,
       {
         timeoutMessage: hovers =>
-          'Timed out waiting for both hovers on valueNode, got hovers:' + JSON.stringify(hovers, undefined, 2)
+          `Timed out waiting for both hovers on ${jsonpathStr} valueNode, got hovers:` +
+          JSON.stringify(hovers, undefined, 2)
       }
     );
-    expect(hovers, 'valueNode hovers').to.not.be.undefined;
+    expect(hovers, `${jsonpathStr} valueNode hovers`).to.not.be.undefined;
     // on the value field, it should have the schema hover and the hover from our provider
-    expect(hovers.length, 'valueNode hovers.length').to.equal(2);
-    if (!hovers.some(h => h.contents.some(c => typeof c === 'object' && c.value.indexOf('StringTypeVar') >= 0))) {
-      expect.fail("Expected at least one hover to contain 'StringTypeVar'");
+    expect(hovers.length, `${jsonpathStr} valueNode hovers.length`).to.equal(2);
+    if (!hovers.some(h => h.contents.some(c => typeof c === 'object' && c.value.indexOf(expectedHoverText) >= 0))) {
+      expect.fail(`Expected at least one ${jsonpathStr} hover to contain '${expectedHoverText}'`);
     }
+  }
+
+  it('hover text on variable names', async () => {
+    const uri = uriFromTestRoot(waveTemplatesUriPath, 'BadVariables', 'layout.json');
+    const [doc] = await openFile(uri, true);
+    await waitForTemplateEditorManagerHas(await getTemplateEditorManager(), uriDirname(uri), true);
+    const tree = parseTree(doc.getText());
+    expect(tree, 'BadVariables/layout.json json tree').to.not.be.undefined;
+    await testHover(doc, uri, tree!, ['pages', 0, 'layout', 'center', 'items', 3, 'name'], 'StringTypeVar');
+    await testHover(
```

Comment:
This bit is refactor to add a test on the variable hovers in GroupBox items, too.